be/src/storage/memtable.cpp

```diff
@@ -24,43 +24,48 @@ namespace starrocks::vectorized {
 static const string LOAD_OP_COLUMN = "__op";
 static const size_t kPrimaryKeyLimitSize = 128;
 
+Schema MemTable::convert_schema(const TabletSchema* tablet_schema, const std::vector<SlotDescriptor*>* slot_descs) {
+    Schema schema = std::move(ChunkHelper::convert_schema_to_format_v2(*tablet_schema));
+    if (tablet_schema->keys_type() == KeysType::PRIMARY_KEYS && slot_descs->back()->col_name() == LOAD_OP_COLUMN) {
+        // load slots have __op field, so add to _vectorized_schema
+        auto op_column = std::make_shared<starrocks::vectorized::Field>((ColumnId)-1, LOAD_OP_COLUMN,
+                                                                        FieldType::OLAP_FIELD_TYPE_TINYINT, false);
+        op_column->set_aggregate_method(OLAP_FIELD_AGGREGATION_REPLACE);
+        schema.append(op_column);
+    }
+    return schema;
+}
+
 void MemTable::_init_aggregator_if_needed() {
     if (_keys_type != KeysType::DUP_KEYS) {
         // The ChunkAggregator used by MemTable may be used to aggregate into a large Chunk,
         // which is not suitable for obtaining Chunk from ColumnPool,
         // otherwise it will take up a lot of memory and may not be released.
-        _aggregator = std::make_unique<ChunkAggregator>(&_vectorized_schema, 0, INT_MAX, 0);
+        _aggregator = std::make_unique<ChunkAggregator>(_vectorized_schema, 0, INT_MAX, 0);
     }
 }
 
-MemTable::MemTable(int64_t tablet_id, const TabletSchema* tablet_schema, const std::vector<SlotDescriptor*>* slot_descs,
+MemTable::MemTable(int64_t tablet_id, Schema* schema, const std::vector<SlotDescriptor*>* slot_descs,
                    MemTableSink* sink, MemTracker* mem_tracker)
         : _tablet_id(tablet_id),
-          _tablet_schema(tablet_schema),
+          _vectorized_schema(schema),
           _slot_descs(slot_descs),
-          _keys_type(tablet_schema->keys_type()),
+          _keys_type(schema->keys_type()),
           _sink(sink),
           _aggregator(nullptr),
           _mem_tracker(mem_tracker) {
-    _vectorized_schema = std::move(ChunkHelper::convert_schema_to_format_v2(*tablet_schema));
```

Comment:
convert_schema_to_format_v2 is useless. We can remove the code also to eliminate the bad case.

Reply:
lator i will submit a pr to remove all the convert_schema_to_format_v2